### PR TITLE
feat: contract call offchain data

### PR DESCRIPTION
**Pull request by frenzox**
**Describe the changes**

Related to https://github.com/eigerco/solana-axelar-internal/issues/526

Added REST Service with endpoint aimed at supporting `CallContract` messages with large payloads that cannot be sent directly on Solana.

- [x] Tests have been added/updated for the changes.
- [x] Documentation has been updated for the changes (if applicable).
- [x] The code follows Rust's style guidelines.

### CODE REVIEW

File: crates/rest-service/src/endpoints/mod.rs
Comment (frenzox) [inline crates/rest-service/src/endpoints/mod.rs:1]:
```suggestion
mod call_contract_offchain_data;
```

File: crates/solana-axelar-relayer/src/main.rs
Comment (frenzox) [inline crates/solana-axelar-relayer/src/main.rs:141]:
```suggestion
```

File: crates/rest-service/src/component.rs
Comment (frenzox) [inline crates/rest-service/src/component.rs:73]: Need to change this as the math is not right

File: crates/rest-service/src/endpoints/call_contract_offchain_data.rs
Comment (roberts-pumpurs) [inline crates/rest-service/src/endpoints/call_contract_offchain_data.rs:80]: misc:
add some logging.
How I'd approach it:
- create a span that captures the signature from the request params
- emit an `info!` log if tx was found and succesfully forwarded, `warn!` log otherwise

**Review by roberts-pumpurs — APPROVED** [review]
lgtm, but don't merge before https://github.com/eigerco/axelar-solana-relayer/pull/2 gets merged

**Comment by frenzox** [comment]
Before merging this we need to update the `axelar-solana` dependecy after https://github.com/eigerco/solana-axelar-internal/pull/568 gets merged and made public.

File: crates/rest-service/src/endpoints/call_contract_offchain_data.rs
Comment (roberts-pumpurs) [reply crates/rest-service/src/endpoints/call_contract_offchain_data.rs:68]: Looks great 🚀

**Review by roberts-pumpurs — DISMISSED** [review]
lgtm, learned some cool things as well 🧠

**Pull request merged**
